gravatar/src/main/java/com/gravatar/services/AvatarService.kt

```diff
@@ -28,15 +28,28 @@ public class AvatarService(private val okHttpClient: OkHttpClient? = null) {
      *
      * @param file The image file to upload
      * @param oauthToken The OAuth token to use for authentication
+     * @param hash The hash of the email to associate the avatar with.
+     * If null the primary email of the account will be used.
+     * @param selectAvatar Determines if the uploaded image should be set automatically as the avatar
+     * for the given hash. If null, the avatar will be selected only if no other avatar is set.
      */
-    public suspend fun upload(file: File, oauthToken: String): Avatar = runThrowingExceptionRequest {
+    public suspend fun upload(
+        file: File,
+        oauthToken: String,
+        hash: Hash? = null,
+        selectAvatar: Boolean? = null,
+    ): Avatar = runThrowingExceptionRequest {
         withContext(GravatarSdkDI.dispatcherIO) {
             val service = instance.getGravatarV3Service(okHttpClient, oauthToken)
 
             val filePart =
                 MultipartBody.Part.createFormData("image", file.name, file.asRequestBody())
 
-            val response = service.uploadAvatar(filePart)
+            val response = service.uploadAvatar(
+                data = filePart,
+                selectedEmailHash = hash.toString(),
```

Comment:
⚠️ We need to use a safe call here. If not, when the hash is `null`, `toString` will return the `"null"` string.
```suggestion
                selectedEmailHash = hash?.toString(),
```

Reply:
💡 Maybe we can add a unit test for that?

Reply:
Good catch! I will add a test 👍

Reply:
Done!